charts/test-guide/templates/config-file.yaml

```diff
@@ -19,23 +19,65 @@ stringData:
     # Database connection
     database.driver={{ .Values.db.driver }}
     database.url={{ .Values.db.url }}
-    database.user={{ .Values.db.user }}
-    database.password={{ .Values.db.password }}
+    database.user=
+    {{- $secret := (lookup "v1" "Secret" .Release.Namespace .Values.secrets.dbCredentials.secretName) -}}
+    {{- if $secret.data }}
+      {{- $dbUser := index $secret.data .Values.secrets.dbCredentials.userKey }}
+      {{- $dbUser }}
```

Comment:
* could be done within one line (just `index ...`)
* you still need to decode 

(applies to all occurrences)